web-common/src/features/dashboards/pivot/PivotTable.svelte

```diff
@@ -44,15 +52,38 @@
   const table = createSvelteTable(options);
 
   let containerRefElement: HTMLDivElement;
+  let stickyRows = [0];
 
   $: assembled = $pivotDataStore.assembled;
   $: expanded = $dashboardStore?.pivot?.expanded ?? {};
   $: sorting = $dashboardStore?.pivot?.sorting ?? [];
-  // $: columnPage = $dashboardStore.pivot.columnPage;
-  // $: totalColumns = $pivotDataStore.totalColumns;
 
   $: headerGroups = $table.getHeaderGroups();
   $: measureCount = $dashboardStore.pivot?.columns?.measure?.length ?? 0;
+  $: rows = $table.getRowModel().rows;
+  $: totalHeaderHeight = headerGroups.length * HEADER_HEIGHT;
+
+  $: virtualizer = createVirtualizer<HTMLDivElement, HTMLTableRowElement>({
+    count: rows.length,
+    getScrollElement: () => containerRefElement,
+    estimateSize: () => ROW_HEIGHT,
+    overscan: OVERSCAN,
+    rangeExtractor: (range) => {
+      const next = new Set([...stickyRows, ...defaultRangeExtractor(range)]);
+
+      return [...next].sort((a, b) => a - b);
+    },
+  });
+
+  $: virtualRows = $virtualizer.getVirtualItems();
+  $: totalRowSize = $virtualizer.getTotalSize();
+
+  $: [before, after] = virtualRows.length
+    ? [
+        (virtualRows[1]?.start ?? virtualRows[0].start) - ROW_HEIGHT,
+        totalRowSize - virtualRows[virtualRows.length - 1].end,
+      ]
+    : [0, 0];
```

Comment:
Would be nice to add some description of what this does and why do we need it

Reply:
Added